### ISSUE TITLE
Angular 2 support

### ISSUE THREAD

**Issue by MedAnd** [bbaia/protractor-net#39]
Just wondering what the status or plans are for Angular 2 support?
ngDriver throws Angular could not be found on the page exception

**Comment by bbaia** [comment]
I'm working on it. Should be available in may.
Stay tuned !

**Comment by MedAnd** [comment]
Nice :+1: Let me know if you need beta testers...

**Comment by lexith** [comment]
+1 Any update on this?

**Comment by bbaia** [comment]
The `$$testability` service is not fully implemented in Angular 2. `NgBy` locators can't be supported for now.
I'll release a version in the next few days to support Angular 2 when waiting for stability or loading a page.

**Comment by claycephas** [comment]
Will this include support for Angular 2 using [ngUpgrade](https://angular.io/docs/ts/latest/guide/upgrade.html)?

**Comment by mlegendre** [comment]
Any update on angular 2 support. I am still running into errors. Whenever I call WaitForAngular method I get angular could not be found on the window.

**Comment by bbaia** [comment]
@mlegendre : Did you tried latest version 0.8 ?

**Comment by mlegendre** [comment]
Yes I have tried your latest.

**Comment by alexkaratzas** [comment]
Any idea when Angular 2 will be fully supported?

**Comment by bbaia** [comment]
What's not working ?
For Angular 2 apps, the binding and model locators are not supported.

**Comment by xcyroo** [comment]
I encounter many TimeOut Exceptions and InvalidStateExceptions running tests against Angular2 app. It works well against Angular 1 app. Any suggestion what could be an issue?

**Comment by xcyroo** [comment]
Does latest version work with angular2 app for someone?

**Comment by bbaia** [comment]
Except NgBy locators that are not supported, it works fine for me.
Are you mixing both angular 1 and 2 version ?

**Comment by xcyroo** [comment]
We are purely on latest version 2 release. I am willing to share more detail the way you prefer.

**Comment by bbaia** [comment]
Is your site online ?
If yes, you can send me the url @ brbaia on gmail.

**Comment by alexkaratzas** [comment]
I am also encountering asynchronous script timeouts trying to test a hybrid 1/2 app. Is this scenario supported?

I am using Protractor.Net v0.9.1 and when I run any test on my hybrid app I get the error below.

Last value checked was OpenQA.Selenium.WebDriverTimeoutException: asynchronous script timeout: result was not received in 5 seconds
  (Session info: chrome=51.0.2704.0)
  (Driver info: chromedriver=2.21.371459 (36d3d07f660ff2bc1bf28a75d1cdabed0983e7c4),platform=Windows NT 6.1 SP1 x86_64)
   at OpenQA.Selenium.Remote.RemoteWebDriver.UnpackAndThrowOnError(Response errorResponse)
   at OpenQA.Selenium.Remote.RemoteWebDriver.Execute(String driverCommandToExecute, Dictionary`2 parameters)
   at OpenQA.Selenium.Remote.RemoteWebDriver.ExecuteScriptCommand(String script, String commandName, Object[] args)
   at OpenQA.Selenium.Remote.RemoteWebDriver.ExecuteAsyncScript(String script, Object[] args)
   at Protractor.NgWebDriver.ExecuteAsyncScript(String script, Object[] args) in c:\Users\Bruno\Projets\GitHub\bbaia\protractor-net\src\Protractor\NgWebDriver.cs:line 434
   at Protractor.NgWebDriver.WaitForAngular() in c:\Users\Bruno\Projets\GitHub\bbaia\protractor-net\src\Protractor\NgWebDriver.cs:line 363
   at Protractor.NgWebDriver.OpenQA.Selenium.ISearchContext.FindElements(By by) in c:\Users\Bruno\Projets\GitHub\bbaia\protractor-net\src\Protractor\NgWebDriver.cs:line 317

As you can see the problem seems to be on WaitForAngular and the observed behaviour is that it is waiting for something and the page never loads. If I refresh the browser after the test times out then I can load the page correctly.

Any ideas what could be causing this or are hybrid apps not supported?

**Comment by bbaia** [comment]
hybrid 1/2 apps are not supported yet

**Comment by RamamurthyB** [comment]
NGWebDriver for Angular 2 is displaying "Driver.url displayed exception of type System.Invalid OperationException
IWebDriver is working fine for Angular 2
NGWebDriver for AngularJS is working fine

            IWebDriver _webDriver = new ChromeDriver(@"C:\Automation\Installs");            
           _webDriver.Navigate().GoToUrl("xxxxxxxxx");
            ngDriver = new NgWebDriver(_webDriver);
            NgWebElement btnApply = ngDriver.FindElement(By.CssSelector("button[type=submit]"))

**Comment by RamamurthyB** [comment]
Is there any settings/config for making Protractor to work for Angular 2 ?

**Comment by bbaia** [comment]
@RamamurthyB you should switch line 2 & 3 : 

```csharp
IWebDriver _webDriver = new ChromeDriver(@"C:\Automation\Installs");            
var ngDriver = new NgWebDriver(_webDriver);
ngDriver.Navigate().GoToUrl("xxxxxxxxx");
NgWebElement btnApply = ngDriver.FindElement(By.CssSelector("button[type=submit]"))
```

**Comment by RamamurthyB** [comment]
Thank you so much Sir! This works!!!!

**Comment by bbaia** [comment]
Just released v0.10 with better Angular 2 support (hybrid apps too)
